PetAdoptions/petsite/petsite/Controllers/BaseController.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http;
+
+namespace PetSite.Controllers
+{
+    public class BaseController : Controller
+    {
+        private static readonly List<string> UserIds = new List<string>
+        {
+            "user001", "user002", "user003", "user004", "user005",
+            "user006", "user007", "user008", "user009", "user010",
+            "user011", "user012", "user013", "user014", "user015",
+            "user016", "user017", "user018", "user019", "user020",
+            "user021", "user022", "user023", "user024", "user025"
+        };
+        private static readonly Random Random = new Random();
+
+        protected bool EnsureUserId()
+        {
+            string userid = Request.Query["userid"];
+            
+            if (string.IsNullOrEmpty(userid))
+            {
+                userid = HttpContext.Session.GetString("userid");
+                if (string.IsNullOrEmpty(userid))
+                {
+                    userid = UserIds[Random.Next(UserIds.Count)];
+                    HttpContext.Session.SetString("userid", userid);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Session fixation vulnerability detected due to session modification before cleanup in authentication context. 
This allows attackers to hijack authenticated sessions by reusing existing session identifiers. To remediate this, always abandon or clear the existing session immediately after successful authentication and before setting any new session values. Use either `HttpContext.Session.Abandon()` or `HttpContext.Session.Clear()` method. More Info: https://owasp.org/www-community/attacks/Session_fixation